cudaaligner/src/alignment_impl.cpp

```diff
@@ -78,31 +78,38 @@ FormattedAlignment AlignmentImpl::format_alignment() const
 {
     int64_t t_pos = 0;
     int64_t q_pos = 0;
-    std::string t_str;
-    std::string q_str;
+    FormattedAlignment ret_formatted_alignment;
+
     for (auto const& x : alignment_)
     {
         switch (x)
         {
         case AlignmentState::match:
+            ret_formatted_alignment.second += target_[t_pos++];
+            ret_formatted_alignment.first += query_[q_pos++];
+            ret_formatted_alignment.pairing += "|";
```

Comment:
Since you only append a single character I would I would use `'` instead of `"`.